### DISCUSSION TITLE
LiquidCrystal Library

### DISCUSSION THREAD

**Discussion by arnab-borua** [SpenceKonde/ATTinyCore#574] [Support / Q & A]
Is the LiquidCrystal Library of Arduino compatible with ATTinyCore? I am using ATTiny84A without bootloader

**Comment by SpenceKonde** [comment]
Which of the thousands of versions of LiquidCrystal.h, which each implement one out of several possible API's are you referring to? LiquidCrystal.h/LiquidCrystal_I2C.h is the library that first made me recognize the dark side of github: Fork proliferation or "fork hell" (by analogy with "DLL hell" on windows, where different programs require different versions of a certain .dll file)

Even a known version of LiquidCrystal is a crufty unmaintained library.

My understanding is that all the cool kids these days are using this one https://github.com/duinoWitchery/hd44780 - this implements a superset of the LiquidCrystal API - while I don't have that much experience working with it, my impression is that it is strictly better than LiquidCrystal.

If hd44780 installed through board manager per the instructions above fails to compile or can't be made to work, that is a bug, please open an issue if that is found to be the case. 

I expect that the io classes which use Wire.h for I2C operation will ALSO work, not just the parallel configuration! Even on parts that don't have proper hardware I2C, ATTinyCore is supposed to paper over that for you so you can #include <Wire.h>, it will use the right implementation for the hardware, and whatever is working with it will have no idea that it's actually using a USI, not a TWI  module. It's not perfect, but it's pretty damned good on the USI parts.
